### PR TITLE
rewritten the multi module/single module gradle jacoco build/code cov…

### PR DESCRIPTION
**Pull request by nigurr**
Enable Jacoco Code coverage support for Gradle Multi Module and Single Module projects.
Fixed issues in Setting the right build paths for Gradle builds

### CODE REVIEW

File: Tasks/Gradle/Gradle.ps1 [hunk 1]
```diff
@@ -89,7 +89,8 @@ if ($jdkPath)
     Write-Verbose "JAVA_HOME set to $env:JAVA_HOME"
 }
 
-$buildRootPath = Split-Path $wrapperScript -Parent
+$buildRootPath = $cwd
+$wrapperDirectory = Split-Path $wrapperScript -Parent
```
Comment (AshwiniChalla) [inline Tasks/Gradle/Gradle.ps1]: Aren't buildRootPath and wrapperDirectory the same?

File: Tasks/Gradle/Gradle.ps1 [hunk 2]
```diff
@@ -98,11 +99,20 @@ $summaryFile = Join-Path $buildRootPath $reportDirectoryName
...
 
+# check if project is multi module gradle build or not
+$subprojects = Invoke-BatchScript -Path $wrapperScript -Arguments 'properties' -WorkingFolder $buildRootPath | Select-String '^subprojects: (.*)'|ForEach-Object {$_.Matches[0].Groups[1].Value}
+Write-Verbose "subprojects: $subprojects"
+$singlemodule = [string]::IsNullOrEmpty($subprojects) -or $subprojects -eq '[]'
+
 # check if code coverage has been enabled
 if($isCoverageEnabled)
 {
+   #Write-Error "cwd $cwd"
```
Comment (AshwiniChalla) [inline Tasks/Gradle/Gradle.ps1]: remove these lines

File: Tasks/Gradle/Gradle.ps1 (hunk 1 above)
Comment (nigurr) [inline Tasks/Gradle/Gradle.ps1]: May not be same

Comment (nigurr) [inline Tasks/Gradle/Gradle.ps1]: You can give build for sub-modules only. However the gradlew exist on main module directory

File: Tasks/Gradle/task.json [hunk 3]
```diff
@@ -80,7 +80,7 @@
             "name": "testResultsFiles",
             "type": "filePath",
             "label": "Test Results Files",
-            "defaultValue": "**/TEST-*.xml",
+            "defaultValue": "**/build/test-results/TEST-*.xml",
```
Comment (acesiddhu) [inline Tasks/Gradle/task.json]: FYI to David

**Comment by acesiddhu** [comment]
looks fine

**Comment by kaadhina** [comment]
Hope you will port this to releases/M91 as well

**Pull request merged**